src/vstart.sh

```diff
@@ -1130,6 +1148,8 @@ EOF
         run 'mgr' $name $CEPH_BIN/ceph-mgr -i $name $ARGS
     done
 
+    wait_for_mgr
```

Comment:
This is better but I don't think it needs to be this verbose or fail if the manager doesn't become available.  A bit lower, there is a loop waiting for the dashboard module which just sleeps for one second without any cutoff.  I think the same should be done here and `wait_for_mgr` can be dropped:
```suggestion
    while ! ceph_adm mgr stat | jq -e '.available'; do
        debug echo 'waiting for mgr to become available'
        sleep 1
    done
```

Reply:
But how about situation that mgr can't start anyway, would this always in the loop?

Reply:
Yes, I don't see a problem with that.  We can't realistically time out every single step in this script.

Reply:
Please don't mark conversations resolved without incorporating appropriate changes or leaving an explicit "I'm resolving this because ..." comment.

Reply:
Sorry to mark this resolved. IMO mgr is an important daemon for cluster, it is necessary to check its status before cluster starts, and this is what real deployment do.

Reply:
This is exactly what the change suggested in this thread does: the script wouldn't proceed until mgr becomes available.

Reply:
Correct!

Reply:
It sounds like you are agreeing to drop `wait_for_mgr` function then?

Reply:
Yes, I agree with you. I keep checking mgr status but drop exiting script.